File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    //Sort the array to ascending order (min -> max)
    nums.sort((a,b)=>a-b);
    //Eliminate the duplication element of the array 
    const set = new Set(nums);
    const uniqueArr = [...set];
    let lengthOfCon = 1; //Variable for checking the length of the longest consecutive elements
    let max =1;
    //Check the uniqueArr is the empty or not 
    if(uniqueArr.length === 0){
        return 0;
    }
    //If the uniqueArr is not empty,
    //Check the next element is the same as the element of the uniqueArr plus one 

    for(let i=1;i<uniqueArr.length;i++){
        if(uniqueArr[i] > uniqueArr[i-1] && Math.abs(uniqueArr[i]-uniqueArr[i-1]) ===1){
                lengthOfCon++;
                max = Math.max(max,lengthOfCon);
        }else if(uniqueArr[i] === uniqueArr[i-1]){
            continue;
        }
        else{
            lengthOfCon = 1;
        }
    }

    return max;


    

    
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) ==0:
            return 0
        #중복제거 set 
        nums = list(set(nums))
        #sort 
        nums.sort()    
        print(nums)

        #Variables
        longest = 1
        current = 1

        #Approah all element of nums for checking the sequnece number or not
        for i in range(1,len(nums)):
            if nums[i] == nums[i-1] + 1:
                current +=1
                longest = max(longest, current) #최대길이 업데이트 
            else:#연속 되지 않을 시 current 1로 초기화
                current =1
        return longest
